autoendpoint/src/routers/fcm/client.rs

```diff
@@ -138,18 +154,24 @@ impl FcmClient {
 
         // Handle error
         let status = response.status();
-        if status.is_client_error() || status.is_server_error() {
-            let data: FcmResponse = response
-                .json()
-                .await
-                .map_err(FcmError::DeserializeResponse)?;
-
-            return Err(match (status, data.error) {
-                (StatusCode::UNAUTHORIZED, _) => RouterError::Authentication,
+        let data: GcmResponse = response
+            .json()
+            .await
+            .map_err(FcmError::DeserializeResponse)?;
+        if status.is_client_error() || status.is_server_error() || data.failure > 0 {
+            return Err(match (status, &data.results[0].error) {
```

Comment:
`results[0]` could panic here if GCM ever sent us bad JSON (e.g. an empty array), let's guard against that.

Reply:
kinda/sorta tempted to say that if we get a panic here, it's indicative of a bigger problem we should be aware of, but yeah, I'm fine handling it.